src/Utils/useMediaDevicePermission.ts

```diff
@@ -21,11 +34,12 @@ export const useMediaDevicePermission = () => {
           return { hasPermission: false, mediaStream: null };
         }
 
-        return { hasPermission: true, mediaStream: mediaStream };
+        return { hasPermission: true, mediaStream };
       } catch (_error) {
         if (!toastShownRef.current) {
           toastShownRef.current = true;
-          toast.warning(t("camera_permission_denied"));
+          const deviceType = options.video ? "camera" : "microphone";
+          toast.warning(t(`${deviceType}_permission_denied`));
```

Comment:
static analysis tools would still find it difficult to detect missing translation coverage on these lines.

```suggestion
          const deviceType = options.video ? "camera" : "microphone";
          toast.warning(options.video ? t("camera_permission_denied") : t("mic_permission_denied"));
```